src/4D_api.cpp

```diff
@@ -966,6 +966,423 @@ std::string pj_add_type_crs_if_needed(const std::string& str)
+ * @param out_top The output value for the top bounding coordinate.
+ * @param densify_pts Recommended to use 21. This is the number of points
+ *     to use to densify the bounding polygon in the transformation.
+ * @return an integer. 1 if successful. 0 if failures encountered.
+ * @since 8.2
+ */
+int proj_trans_bounds(PJ *P,
+                      PJ_DIRECTION direction,
+                      const double left,
+                      const double bottom,
+                      const double right,
+                      const double top,
+                      double* out_left,
+                      double* out_bottom,
+                      double* out_right,
+                      double* out_top,
+                      int densify_pts
+) {
+    *out_left = HUGE_VAL;
+    *out_bottom = HUGE_VAL;
+    *out_right = HUGE_VAL;
+    *out_top = HUGE_VAL;
+
+    if (P == nullptr) {
+        proj_log_error(P, _("NULL P object not allowed,"));
+        proj_errno_set (P, PROJ_ERR_INVALID_OP_ILLEGAL_ARG_VALUE);
+        return false;
+    }
+    if (densify_pts < 0) {
+        proj_log_error(P, _("densify_pts must be greater than 0."));
+        proj_errno_set (P, PROJ_ERR_INVALID_OP_ILLEGAL_ARG_VALUE);
+        return false;
+    }
+
+    PJ_PROJ_INFO pj_info = proj_pj_info(P);
+    if (pj_info.id == nullptr) {
+        proj_log_error(P, _("NULL transformation not allowed,"));
+        proj_errno_set (P, PROJ_ERR_INVALID_OP_ILLEGAL_ARG_VALUE);
+        return false;
+    }
+    if (strcmp(pj_info.id, "noop") == 0) {
+        *out_left = left;
+        *out_right = right;
+        *out_bottom = bottom;
+        *out_top = top;
+        return true;
+    }
+
+    bool degree_output = proj_degree_output(P, direction) != 0;
+    bool degree_input = proj_degree_input(P, direction) != 0;
+    if (degree_output && densify_pts < 2) {
+        proj_log_error(P, _("densify_pts must be at least 2 if the output is geograpic."));
+        proj_errno_set (P, PROJ_ERR_INVALID_OP_ILLEGAL_ARG_VALUE);
+        return false;
+    }
+
+    int side_pts = densify_pts + 1;  // add one because we are densifying
+    int boundary_len = side_pts * 4;
+    std::vector<double> x_boundary_array(boundary_len);
+    std::vector<double> y_boundary_array(boundary_len);
+    double delta_x = 0;
+    double delta_y = 0;
+    bool north_pole_in_bounds = false;
+    bool south_pole_in_bounds = false;
+
+    if (degree_output) {
+        north_pole_in_bounds = contains_north_pole(
+            P,
+            direction,
+            left,
+            bottom,
+            right,
+            top
+        );
+        south_pole_in_bounds = contains_south_pole(
+            P,
+            direction,
+            left,
+            bottom,
+            right,
+            top
+        );
+    }
+
+    if (degree_input && right < left) {
+        // handle antimeridian
+        delta_x = (right - left + 360.0) / ( (double) side_pts );
```

Comment:
implict cast from int to double will be done here
```suggestion
        delta_x = (right - left + 360.0) / side_pts;
```
applies for below similar occurences

Reply:
Good to know, I will remove it.